### ISSUE TITLE
Example from readme doesn't build

### ISSUE THREAD

**Issue by MichaelSenko** [NLua/NLua#364]
I'm attempting to convert a project from the old LuaInterface to NLua, mostly for easier 64-bit support.  I've run into a couple of items that worked OK with LuaInterface but now cause troubles with NLua.  First item is converting Lua values back into C# native types.  One of the examples from the readme demonstrates the problem I've encountered:

            Lua state = new Lua();
            double val = 12.0;
            state["x"] = val;
            var res = state.DoString("return 10 + x*(5_2)");
            state.DoString("y=10+x*(5+2)");
            var y = state["y"] as double;

The last line results in a build error:

Severity	Code	Description	Project	File	Line	Suppression State
Error	CS0077	The as operator must be used with a reference type or nullable type ('double' is a non-nullable value type

I'm coming at this from the Lua side and have limited experience with .NET.  The only mechanism I've found to convert Lua numbers back into C# doubles or int's is to call the .ToString() method, and then Parse() back into the numeric format.  Example below works fine, but seems like a hack.  Am I missing something obvious?  I see the GetNumber() method for the Lua state, but that seems to only work with globals, and not for example if I have a C# function that takes a LuaTable as a parameter.

    ...
   state.DoString("y=10+x*(5+2)");
   double y = double.Parse(state["y"].ToString());

Thanks
Mike

**Comment by viniciusjarina** [comment]
You are right, the `as` expression will not work, but the normal cast will.
Feel free to send a PR fixing those errors.

Thank you @MichaelSenko

**Comment by MichaelSenko** [comment]
Sent a PR.  Git newbie, so a bit sloppy.

The cast does work for doubles, but not for integers.

state.DoString("y=10*(5+2)");
int y = (int) state["y"];

This will build, but results in a runtime error.  Is this due to the Lua 5.3 distinction between numbers and integers?

Thanks
Mike
